### PR TITLE
LNK-2722: Implement repository pattern in the Audit service

### PR DESCRIPTION
**Pull request by amphillipsLGC**
- [x] Create new Audit Manager
- [x] Refactor commands and queries to follow new repository pattern
   - [x] `CreateAuditEvent` command updated
   - [x] `GetAuditEvent` query updated
   - [x] `GetAuditEventList` query updated
- [x] Update Audit Event Listener to use the Audit Manager
- [x] Refactor factory logic into record classes
- [x] Update Unit Tests

## Summary by CodeRabbit

- **New Features**
	- Introduced a new interface and class for processing audit events.
	- Added a new constructor for flexible initialization of audit search filters.

- **Improvements**
	- Enhanced the structure and clarity of the audit logging process.
	- Streamlined the controller to improve performance and reduce complexity.
	- Improved error handling and service registration processes.

- **Bug Fixes**
	- Ensured better handling of audit event logging and retrieval processes.

- **Tests**
	- Added unit tests for the newly implemented audit event processing functionality.
	- Updated the testing framework from NUnit to xUnit for improved compatibility and functionality.

### CODE REVIEW

**Review by nvmLantana — APPROVED** [review]
Looks good to meeeee. nice to see so many cleaned up files!

EDIT: I take it back! lol

**Review by nvmLantana — COMMENTED** [review]
Address build failures and retest?

File: DotNet/Audit/Infrastructure/Logging/Logging.cs [hunk 1]
```diff
@@ -1,5 +1,4 @@
-﻿using LantanaGroup.Link.Audit.Application.Commands;
-using LantanaGroup.Link.Audit.Application.Models;
+﻿using LantanaGroup.Link.Audit.Application.Models;
```
[the remark this answers is not in the document]
Comment (amphillipsLGC) [reply DotNet/Audit/Infrastructure/Logging/Logging.cs:1]: that import was removed

**Review by nvmLantana — APPROVED** [review]
Looks Good

**Pull request merged**